Lib/Application.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace Core;
+
+class Application
+{
+
+    /**
+     * @var HTTPRequest
+     */
+    protected $httpRequest;
+    /**
+     * @var HTTPResponse
+     */
+    protected $httpResponse;
+    /**
+     * @var
+     */
+    protected $env = 'Frontend';
+    /**
+     * @var
+     */
+    protected $name;
+    /**
+     * @var
+     */
+    protected $config;
+
+    public function __construct()
+    {
+        $this->httpRequest  = new HTTPRequest($this);
```

Comment:
Question : pourquoi cette référence circulaire ? (le this qui pointe sur l'applicatiion)

Reply:
La réponse : placé l'instance de l'application dans l'objet rend impossible une instanciation en dehors de l'objet en question.